bin/cli.js

```diff
@@ -197,13 +199,30 @@ function invoke(env) {
       `-x [${filetypes.join('|')}]`,
       'Specify the stub extension (default js)'
     )
+    .option(
+      `--stub [<relative/path/from/knexfile> | <name>]`,
+      'Specify the migration stub to use. If using <name> the file must be located in config.migrations.directory'
+    )
     .action((name) => {
       const opts = commander.opts();
       opts.client = opts.client || 'sqlite3'; // We don't really care about client when creating migrations
       const instance = initKnex(env, opts);
       const ext = getMigrationExtension(env, opts);
+      let { stub } = argv;
+      if (stub) {
```

Comment:
I would prefer to encapsulate this logic inside a separate function, akin to `const ext = getMigrationExtension(env, opts);`.
And, TBH, probably both of them could be moved to `bin/utils/cli-config-utils.js`.

Reply:
sure thing. i wrote it both ways in the original issue so ill push it up with the util function from there for you to take a look.